sdk/inc/azure/core/az_mqtt5_connection.h

```diff
@@ -154,6 +154,12 @@ struct az_mqtt5_connection
      */
     az_mqtt5_connection_callback event_callback;
 
+    /**
+     * @brief Context for MQTT 5 connection events callback.
+     *
+     */
+    const void* event_callback_context;
```

Comment:
since there isn't an example in the samples - what is this expected to be used for?

Reply:
The use case is when a customer wants to add this SDK into their solution and they do not want to define global variables to access the application state. They want to invoke the _init function from main and receive a reference to their app state as a context in the callback.